reana_workflow_controller/rest.py

```diff
@@ -918,7 +921,125 @@ def get_workflow_status(workflow_id):  # noqa
+          required: true
+          type: string
+        - name: status
+          in: body
+          description: Required. New status.
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: >-
+            Request succeeded. Info about workflow, including the status is
+            returned.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+              organization:
+                type: string
+              status:
+                type: string
+              user:
+                type: string
+          examples:
+            application/json:
+              {
+                "id": "256b25f4-4cfb-4684-b7a8-73872ef455a1",
+                "organization": "default_org",
+                "status": "running",
+                "user": "00000000-0000-0000-0000-000000000000"
+              }
+        400:
+          description: >-
+            Request failed. The incoming data specification seems malformed.
+          examples:
+            application/json:
+              {
+                "message": "Malformed request."
+              }
+        403:
+          description: >-
+            Request failed. User is not allowed to access workflow.
+          examples:
+            application/json:
+              {
+                "message": "User 00000000-0000-0000-0000-000000000000
+                            is not allowed to access workflow
+                            256b25f4-4cfb-4684-b7a8-73872ef455a1"
+              }
+        404:
+          description: >-
+            Request failed. Either User or Workflow doesn't exist.
+          examples:
+            application/json:
+              {
+                "message": "User 00000000-0000-0000-0000-000000000000 doesn't
+                            exist"
+              }
+            application/json:
+              {
+                "message": "Workflow 256b25f4-4cfb-4684-b7a8-73872ef455a1
+                            doesn't exist"
+              }
+        500:
+          description: >-
+            Request failed. Internal controller error.
+    """
 
+    try:
+        organization = request.args['organization']
+        user_uuid = request.args['user']
+        workflow = Workflow.query.filter(Workflow.id_ == workflow_id).first()
+        status = request.json
+        if not workflow:
+            return jsonify({'message': 'Workflow {} does not exist'.
+                            format(workflow_id)}), 404
+        if not str(workflow.owner_id) == user_uuid:
+            return jsonify(
+                {'message': 'User {} is not allowed to access workflow {}'
+                 .format(user_uuid, workflow_id)}), 403
+        if status == 'start':
+            return start_workflow(organization, user_uuid, workflow)
         return jsonify({'id': workflow.id_,
                         'status': workflow.status.name,
```

Comment:
This status will be the previous (the one you just got from DB), one would expect to retrieve the requested status. I would return, for now, the requested status as it is (as long as it is correct). 

I know it is misleading (if you want to start, it could perfectly happen that it fails) but since we are calling a celery task which doesn't return any value and until [this](https://github.com/reanahub/reana-workflow-engine-yadage/blob/master/reana_workflow_engine_yadage/tasks.py#L109) gets executed in the workflow engine there won't be any change in DB we can't do anything else but returning the value as it is.

Reply:
I will return the requested status for now, but we could extend the set of statuses and return here "starting" or "pending", for example, and "started" when the workflow starts in the workflow engine.